Initial iteration of the React Map component

mendozaline opened the pull request:
This is the basic map component for issue [#62](https://github.com/hackoregon/civic/issues/62). I added react-map-gl, created a new React Map component and story for it. 

I want to get this reviewed before moving on to the deck.gl overlays.

A testing file still needs to be added and written. I'll need help with that.

Some questions that we still need to resolve include what to do about a default map style and any other defaults we want to set.

cp commented on packages/component-library/src/ReactMap/ReactMap.js:
```diff
@@ -0,0 +1,60 @@
+import MapGL from 'react-map-gl';
+import './mapbox-gl.css';
+
+const ReactMap = (props) => {
```
Might make sense to simplify the name of this component (and file) to just `Map`. Thoughts?

mendozaline replied:
That makes sense. I was trying to differentiate it from the Leaflet Map component already there. I'll rename it.

mendozaline commented:
I renamed the component from ReactMap to BaseMap and converted it from a functional to class component. This allows the map to now pan, zoom, and resize.

I also included and wrote tests for the component.

jaronheard commented on packages/component-library/src/BaseMap/BaseMap.js:
```diff
@@ -0,0 +1,82 @@
+    super(props);
+    this.state = {
+      viewport:{
+        width: window.innerWidth > 900 ? 898 : window.innerWidth,
```
May change this to work with responsive containers if it doesn't already

mendozaline replied:
I was meaning to change that. It's already inside a div and it's responsive, so the map resize when the window changes and stays centered. But I'll update it.
